### DISCUSSION TITLE
Account linking / unlinking

### DISCUSSION THREAD

**Discussion by ScreamZ** [nextauthjs/next-auth#11094] [Help]
Hi folks,

I was looking for a way to programmatically link/unlink accounts to user.

Currently, this is only working if the user is already login to one provider and login to another.
The only way to unlink I found was to delete manually relation in the database.

Is there another way ? To handle in a better way accounts merging

Thank you

**Comment by k-taro56** [comment]
Hi,

What exactly do you want to do?

I would add that the `link` callback function is also called when a user logs in for the first time.
This is because the essence of the `link` callback function is called when the user table and the account table in the database are linked together. (To begin with, it is not a link between multiple providers.)

Thank you!
